flink-core/src/main/java/org/apache/flink/configuration/PipelineOptionsInternal.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration;
+
+import org.apache.flink.annotation.Internal;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+
+/**
+ * Pipeline options that are not meant to be used by the user.
+ */
+@Internal
+public class PipelineOptionsInternal {
+
+	public static final ConfigOption<String> PIPELINE_FIXED_JOB_ID =
```

Comment:
We could also make this a regular public option but describe that it should only be used in special cases. Users will have to use it in the standalone case, where it would be good to also allow configuration of the fixed job id via the flink conf. WDYT?

Reply:
I would prefer to do it in a separate commit/jira because:
1) it seems an important change and
2) what happens in the case of multi-`execute()` applications? All the jobs will have the same id? And if we make it a list, what happens in the case of a non-deterministic `main()`.